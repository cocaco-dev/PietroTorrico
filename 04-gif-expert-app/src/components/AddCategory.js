import React, {useState} from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState(''); //Undefined si no se pone las comillas
    const handleInputChange = (e) => {        
        setInputValue(e.target.value)        
    }
    const handleSubmit = (e) => {
        e.preventDefault();   
        console.log('ENVIO', inputValue);     
        if(inputValue.trim().length > 2){
            setCategories(cate => [inputValue, ...cate]); //cate hace referencia al contenido del componente  
            setInputValue('');
        }
    }

    return (
        <form onSubmit = {handleSubmit}>        
            {/* <p>{inputValue}</p>     */}
            <input
                type="text"
                value = {inputValue}
                onChange={handleInputChange}
                placeholder = 'Ingresa una categoria'
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}