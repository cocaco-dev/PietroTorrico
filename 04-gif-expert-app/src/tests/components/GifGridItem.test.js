import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom'
import { GifGridItem } from '../../components/GifGridItem';

describe('Debemos mostrar el componente <GifGridItem/>', () => {   
    const title = 'Un título';
    const url = "https://google.com";
    const className = "animate__fadeIn";
    const wrapper = shallow(<GifGridItem title = {title} url = {url}/>)

    test('Muestra el componente ', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de tener un parrafo con el title ', () => {
        const p = wrapper.find('p').text();
        expect(p.trim()).toBe(title)
    })

    test('Debe de tener una imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img')
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('Debe tener el className = FadeIn ', () => {
        const div = wrapper.find('div');           
        expect(div.prop('className').includes(className)).toBe(true); //Verificamos que en classname exista la const classname, si es así es tru
    })
    
    
    
})