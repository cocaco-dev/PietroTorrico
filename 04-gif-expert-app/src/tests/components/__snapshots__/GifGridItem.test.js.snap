// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debemos mostrar el componente <GifGridItem/> Muestra el componente  1`] = `
<div
  className="card animate__animated animate__fadeIn "
>
  <p>
    Un título
  </p>
  <img
    alt="Un título"
    src="https://google.com"
  />
</div>
`;
